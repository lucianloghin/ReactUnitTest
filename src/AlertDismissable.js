import React from 'react';
import { Alert, Button, Collapse } from 'react-bootstrap';
import ExtraComponent from './ExtraComponent'

export default class AlertDismissable extends React.Component {
  render() {
    return (
      <Collapse in={this.props.show}>
        <Alert bsStyle="info" onDismiss={this.onDismiss}>
          <h4>
            {this.props.title}
          </h4>
          <p>
            {this.props.message};
          </p>
          <ExtraComponent />
          <p>
            <Button onClick={this.props.onDismiss}>{this.props.dismissText}</Button>
          </p>
        </Alert>
      </Collapse>
    );
  }
}
