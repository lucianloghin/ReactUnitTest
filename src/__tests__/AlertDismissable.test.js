import React from 'react';
import AlertDismissable from "../AlertDismissable";
import renderer from 'react-test-renderer';

test('Alert is rendered correctly', () => {
  const component = renderer
                      .create(
                              <AlertDismissable 
                                title='Error'
                                show={true}
                                message='Unexpected error'
                                dismissText='Close'/>)
                      .toJSON();

  expect(component).toMatchSnapshot();
});

