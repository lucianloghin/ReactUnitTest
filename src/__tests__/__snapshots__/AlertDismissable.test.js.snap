// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert is rendered correctly 1`] = `
<div
  aria-expanded={null}
  bsStyle="info"
  className="fade collapse show alert show"
  role="alert"
>
  <h4>
    Error
  </h4>
  <p>
    Unexpected error
    ;
  </p>
  <div>
    <p>
      This is an extra component that will be rendered
    </p>
  </div>
  <p>
    <button
      className="btn btn-primary"
      disabled={false}
      type="button"
    >
      Close
    </button>
  </p>
</div>
`;
